### ISSUE TITLE
Softmax

### ISSUE THREAD

**Issue by partha2409** [funcwj/conv-tasnet#2]
Hi, i see u have tabulated all results based on Relu or Sigmoid as the non-linear layer. Did you try using softmax?If yes, how were the results? The paper architecture has softmax layer after we get the masks.

**Comment by funcwj** [comment]
Originally I used softmax (the same non-linear function as the paper), but could not get good results (in fact, very bad). The model convergents very slowly and I tune the parameters for nearly two weeks. Finally I used relu instead of softmax, then the result gets better. That's a interesting problem.

**Comment by partha2409** [comment]
Yes i face the same problem. When i use softmax, the loss stops around -0.5 to -1.5 even after 20-30 epochs. however i was able to train the LSTM based tasnet paper in tensorflow using softmax and it converged well. I'm new to pytorch and thought may be i was missing something.Thanks for the confirmation. Even i'm not sure why it happens.

**Comment by funcwj** [comment]
Someone tell me replacing softmax with relu in LSTM tasnet could also bring improvements. Maybe you can check it.

**Comment by partha2409** [comment]
Let me try that as well ! Thanks again :)
